### PR TITLE
Improve schedule wrap docs

### PR DESCRIPTION
**Pull request by taylan97**
I didn't understand that `schedule_wrap`, returns a *function* that you have to call yourself.

The name does actually imply this, _if_ you know what wrap is supposed to mean, but it caused me a lot of confusion and I was on the edge of posting an issue complete with an example `init.vim` before it actually connected.

So I thought maybe the docs for `schedule_wrap` could be a bit more explicit in it's usage, since it's easy to skim the few examples in the docs and not realize the actual nature of it's usage.

English is not my primary tongue so I hope it has been written correctly.

### CODE REVIEW

File: runtime/doc/lua.txt [hunk 1]
```diff
@@ -968,16 +968,35 @@ paste({lines}, {phase})                                          *vim.paste()*
...
+                such as `vim.fn_defer`.
+
+                Example: Using as an argument >
+
+                 local safe = vim.schedule_wrap(function()
+		   vim.api.nvim_command('echomsg "test"')
+                 end)
+                 vim.defer_fn(safe, 100)
+
+                Example: Using on it's own >
```
Comment (clason) [inline runtime/doc/lua.txt:985]: typo: "its"

File: runtime/doc/lua.txt [hunk 2]
```diff
@@ -968,16 +968,35 @@ paste({lines}, {phase})                                          *vim.paste()*
                     |paste|
 
 schedule_wrap({cb})                                      *vim.schedule_wrap()*
-                Defers callback `cb` until the Nvim API is safe to call.
+                Wraps callback `cb` in a function which will defer the execution
+                of `cb` until the Nvim API is safe to call.
+
+                The user must call the returned function themselves if it is not
+                being used as an argument to a function that will call it for them,
+                such as `vim.fn_defer`.
```
Comment (clason) [inline runtime/doc/lua.txt:976]: It's `defer_fn` -- and this function _itself_ wraps the argument in `schedule_wrap`, so mentioning this here is a bit misleading?

Comment (clason) [reply runtime/doc/lua.txt:976]: Maybe it would be better instead to start with the second example and then refer to `defer_fn` as a simple, more "high-level" alternative that defers without requiring wrapping?

Comment (bfredl) [reply runtime/doc/lua.txt:976]: A better example would be to use a raw luv callback or like nvim_buf_attach.

File: runtime/doc/lua.txt (hunk 1 above)
Comment (taylan97) [reply runtime/doc/lua.txt:985]: I followed the capitalization example from other examples,

https://github.com/taylan97/neovim/blob/959349b688c282bbead31067d62507d63d1732e6/runtime/doc/lua.txt#L498

https://github.com/taylan97/neovim/blob/959349b688c282bbead31067d62507d63d1732e6/runtime/doc/lua.txt#L942

Though others do not capitalize after the colon,

https://github.com/taylan97/neovim/blob/959349b688c282bbead31067d62507d63d1732e6/runtime/doc/lua.txt#L480

For datas sake, more examples start with a capital than do not.

```
~/neovim/runtime/doc $ rg --no-heading "Example: [A-Z]" | wc -l
26 (-3, three of these are "TCP", "ISO" and "UTF")

~/neovim/runtime/doc $ rg --no-heading "Example: [a-z]" | wc -l
17
```

(do not fret about the lack of examples, there are 456 examples that simply read  "Example: >")

I do not say this to be pedantic to "win the argument", only to point out perhaps another potential (very minor) doc style fault (can open another pull request but maybe it would conflict too much with upstream vim I don't know how that works).

I will alter it to be lower cased, most online rules seem to say this is most accepted as "correct, or at least 'not incorrect'".

File: runtime/doc/lua.txt (hunk 2 above)
Comment (taylan97) [reply runtime/doc/lua.txt]: My mistake, I did not realize it automatically added it's own wrapping (Z_Z). I agree this is a poor example.

`nvim_buf_attach` is probably a good choice because it may be an exact time you want to use `schedule_wrap`. I'm not quite sure how complicated to make the callback.

For now I have left it as the "echomsg" function. I couldn't think of something that was simple enough to be one line, while still requiring schedule_wrap. My API knowledge isn't very deep.

**Comment by taylan97** [comment]
Corrected example capitalization and re-wrote advanced example to use nvim_buf_attach.

File: runtime/doc/lua.txt (hunk 1 above)
Comment (clason) [reply runtime/doc/lua.txt]: That's fine, no need to worry about it. It is indeed a matter of style (which cannot be expected to be consistent in a document grown over decades). I wouldn't have mentioned it had I not already noticed the typo ;)

File: runtime/doc/lua.txt [hunk 3]
```diff
@@ -968,16 +968,35 @@ paste({lines}, {phase})                                          *vim.paste()*
                     |paste|
 
 schedule_wrap({cb})                                      *vim.schedule_wrap()*
-                Defers callback `cb` until the Nvim API is safe to call.
+                Wraps callback `cb` in a function which will defer the execution
+                of `cb` until the Nvim API is safe to call.
+
+                The user must call the returned function themselves if it is not
+                being used as an argument to a function that will call it for them,
+                such as |nvim_buf_attach()|.
```
Comment (tjdevries) [inline runtime/doc/lua.txt:976]: I think you could say something more along the lines of: "{cb} is not called after being wrapped. To automatically call, use |vim.schedule()|"

Or something like that?

I think this is too wordy.

**Comment by tjdevries** [comment]
Also, you should update the source code, not the API docs for this, as we generate these docs (so your changes will disappear).

If you update the source code, I can push a commit that updates the generated docs for you.

File: runtime/doc/lua.txt (hunk 3 above)
Comment (tjdevries) [reply runtime/doc/lua.txt:976]: I guess I'm trying to say, if you're doing a one-off wrap & call, you can just use `vim.schedule`. If you want to keep the function always safe to call, then you should use `vim.schedule_wrap`. 

I'm too tired to think of a good way to say it thought.

**Comment by Qyriad** [comment]
For future searchers: improvements to these docs were merged in #25286, which at the time of this writing isn't in a stable Neovim release yet.
